File: clients/java/cldf-java/src/main/java/io/cldf/globalid/Sector.java
package io.cldf.globalid;

import lombok.Builder;

/** Represents a sector within a climbing location */
@Builder
public record Sector(String name, Integer order) {

  /** Constructor with just name (order defaults to null) */
  public Sector(String name) {
    this(name, null);
  }
}

File: clients/dart/cldf/lib/clid/models/sector.dart
/// Represents a sector within a climbing location
class Sector {
  final String name;
  final int? order;
  
  const Sector({
    required this.name,
    this.order,
  });
  
  Map<String, dynamic> toJson() => {
    'name': name,
    if (order != null) 'order': order,
  };
  
  factory Sector.fromJson(Map<String, dynamic> json) {
    return Sector(
      name: json['name'] as String,
      order: json['order'] as int?,
    );
  }
}